File: Snooper/src/main/java/com/prateekj/snooper/SnooperShakeListener.java
package com.prateekj.snooper;

public class SnooperShakeListener implements OnShakeListener {

  private SnooperShakeAction shakeAction;
  private boolean isSnooperFlowStarted;

  public SnooperShakeListener(SnooperShakeAction shakeAction) {
    this.shakeAction = shakeAction;
  }

  @Override
  public void onShake() {
    if (!isSnooperFlowStarted) {
      this.shakeAction.startSnooperFlow();
      isSnooperFlowStarted = true;
      return;
    }
    this.shakeAction.endSnooperFlow();
    isSnooperFlowStarted = false;
  }
}

File: Snooper/src/main/java/com/prateekj/snooper/SnooperShakeListener.kt
package com.prateekj.snooper

class SnooperShakeListener(private val shakeAction: SnooperShakeAction) : OnShakeListener {
  private var isSnooperFlowStarted: Boolean = false

  override fun onShake() {
    if (!isSnooperFlowStarted) {
      this.shakeAction.startSnooperFlow()
      isSnooperFlowStarted = true
      return
    }
    this.shakeAction.endSnooperFlow()
    isSnooperFlowStarted = false
  }
}
